### DISCUSSION TITLE
Be careful with ToListAsync and ToArrayAsync in Entity Framework Core

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#121] [General]
# Be careful with ToListAsync and ToArrayAsync in Entity Framework Core

Entity Framework has two methods for converting a query to a list or an array: ToListAsync and ToArrayAsync. Of course there is also the sync version of these methods: ToList and ToArray. And sometimes, you should use the latter one!

https://steven-giesel.com/blogPost/70cebcf5-d171-4852-ae73-4950accfe999/be-careful-with-tolistasync-and-toarrayasync-in-entity-framework-core

**Comment by puschie286** [comment]
Following the github discussion, some improvements were made in the sql client to improve this (might be contained in efCore 9) [PR](https://github.com/dotnet/SqlClient/pull/2164)

**Reply by linkdotnet** [reply]
uhhh nice! Thanks for the update
